### DISCUSSION TITLE
> Python 3.8 support for BACpypes

### DISCUSSION THREAD

**Discussion by parthgoel06** [JoelBender/bacpypes#499] [Q&A]
I've been using BACpypes for BACnet IP and MSTP on a Raspberry Pi running Buster image which has Python 3.7 installed. I recently upgraded to Bullseye Image which has Python 3.9 and things started to break. I checked on pypi.org that BACpypes only supports Python versions till Python 3.8. Is there any development in the pipeline to support Python > 3.8 in the future?

**Comment by JoelBender** [comment]
Yes, [BACpypes3](https://github.com/JoelBender/BACpypes3) is in process, re-written to use [asyncio](https://docs.python.org/3/library/asyncio.html).  I have not approached the author of [misty](https://github.com/riptideio/misty) to understand how to integrate it and I don't have any MSTP devices to test against, so I would appreciate some help with that.  You will probably breeze through the [documentation](https://bacpypes3.readthedocs.io/en/latest/) in short order, and things like `value = await app.read_property(...)` a notable improvement.
